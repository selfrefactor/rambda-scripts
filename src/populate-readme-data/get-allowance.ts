import {EXAMPLE_ONLY_METHODS, METHODS_TO_SKIP, METHODS_TO_SKIP_RAMBDAX} from '../constants'

interface Allowance {
  typing: boolean
  example: boolean
  allTypings: boolean
  source: boolean
  specs: boolean
  tsTest: boolean
  benchmark: boolean
}

export function getAllowance(
  methodName: string,
  withRambdax: boolean,
	docsifyMode: boolean
): Allowance {
  if (withRambdax) {
    const allowed = docsifyMode ? true : !METHODS_TO_SKIP_RAMBDAX.includes(methodName)
		let allTypings = docsifyMode	

    return {
      typing: allowed,
      example: allowed,
      allTypings: allTypings,
      source: allowed,
      specs: allowed,
      tsTest: false,
      benchmark: false,
    }
  }
	const allowed = docsifyMode ? true : !METHODS_TO_SKIP.includes(methodName)

  // const allowed = docsifyMode ? true : !METHODS_TO_SKIP.includes(methodName) && !EXAMPLE_ONLY_METHODS.includes(methodName)
  // const exampleIsAllowed = docsifyMode ? true : EXAMPLE_ONLY_METHODS.includes(methodName)
  
  return {
    typing: allowed,
    example: allowed,
    allTypings: allowed,
    source: allowed,
    specs: allowed,
    tsTest: allowed,
    benchmark: false,
  }
}
