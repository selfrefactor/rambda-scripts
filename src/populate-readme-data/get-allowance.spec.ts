import { getAllowance } from './get-allowance'

describe('happy', () => {
  ['add', 'adjust', `pipe`].forEach((method) => {
    test((method), () => {
      const result =  getAllowance(method, false)
      expect(
        result
      ).toMatchSnapshot()
    })
  })
})
