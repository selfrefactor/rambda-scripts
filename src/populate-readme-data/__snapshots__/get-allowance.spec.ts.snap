// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy add 1`] = `undefined`;

exports[`happy adjust 1`] = `undefined`;

exports[`happy pipe 1`] = `undefined`;
